src/polkit/polkitunixprocess.c

```diff
@@ -376,37 +376,19 @@ polkit_unix_process_constructed (GObject *object)
 {
   PolkitUnixProcess *process = POLKIT_UNIX_PROCESS (object);
 
-  /* sets pidfd, start_time and uid in case they are unset */
+  /* sets start_time and uid in case they are unset */
 
   /* We didn't open it ourselves here, so we must have got it
    * from D-Bus, mark it as safe to use */
   if (process->pidfd >= 0)
     process->pidfd_is_safe = TRUE;
 
 #ifdef HAVE_PIDFD_OPEN
-  if (process->pid > 0 && process->pidfd < 0)
-    {
-      gint pidfd = (int) syscall (SYS_pidfd_open, process->pid, 0);
-      if (pidfd >= 0)
-        {
-          process->pidfd = pidfd;
-          process->pid = 0;
-        }
-    }
-
     if (process->pidfd >= 0)
       {
         GError *error;
         error = NULL;
 
-        process->ppidfd = get_ppidfd_for_pidfd (process->pidfd, &error);
```

Comment:
Same as above, this cannot be dropped. Instead add the same check as above to see if it was already set